File: app/src/test/java/com/mgaetan89/showsrage/adapter/ScheduleAdapter_GetItemCountTest.java
package com.mgaetan89.showsrage.adapter;

import com.mgaetan89.showsrage.model.Schedule;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class ScheduleAdapter_GetItemCountTest {
	@Parameterized.Parameter(0)
	public List<Schedule> schedules;

	@Parameterized.Parameter(1)
	public int itemCount;

	private ScheduleAdapter adapter;

	@Before
	public void before() {
		this.adapter = new ScheduleAdapter(this.schedules);
	}

	@Test
	public void getItemCount() {
		assertThat(this.adapter.getItemCount()).isEqualTo(this.itemCount);
	}

	@After
	public void after() {
		this.adapter = null;
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{Collections.emptyList(), 0},
				{Collections.singletonList(new Schedule()), 1},
				{Arrays.asList(new Schedule(), new Schedule(), new Schedule()), 3},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/adapter/ScheduleAdapter_GetItemCountTest.kt
package com.mgaetan89.showsrage.adapter

import com.mgaetan89.showsrage.model.Schedule
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class ScheduleAdapter_GetItemCountTest(val schedules: List<Schedule>, val itemCount: Int) {
    private lateinit var adapter: ScheduleAdapter

    @Before
    fun before() {
        this.adapter = ScheduleAdapter(this.schedules)
    }

    @Test
    fun getItemCount() {
        assertThat(this.adapter.itemCount).isEqualTo(this.itemCount)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any>> {
            return listOf(
                    arrayOf(emptyList<Any>(), 0),
                    arrayOf(listOf(Schedule()), 1),
                    arrayOf(listOf(Schedule(), Schedule(), Schedule()), 3)
            )
        }
    }
}
